certbot-nginx/certbot_nginx/parser.py

```diff
@@ -28,7 +28,7 @@ class NginxParser(object):
     """
 
     def __init__(self, root):
-        self.parsed = {}
+        self.parsed = {} # type: Dict
```

Comment:
Not sure how much this buys us, but I think we can be a little more specific here. `type: Dict[str, Union[List, nginxparser.UnspacedList]]` should be correct.

@jameshiebert, I'd use the type in my comment for now, but I'm curious what you think @ohemorange.

Reply:
yes, that's correct. hopefully with sydney's changes we can update this to not need a union.